File: crossecore-java/src/java_/ManifestEntryBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class ManifestEntryBase 
extends org.eclipse.emf.ecore.impl.MinimalEObjectImpl.Container implements ManifestEntry
{
	protected static final java.lang.String NAME_EDEFAULT = "";
	private java.lang.String _name = NAME_EDEFAULT;
	public java.lang.String getName()
	{
		return _name;
	}
	public void setName(java.lang.String value){
		
		java.lang.String oldValue = _name;
		_name = value;
		if (eNotificationRequired())
			eNotify(new ENotificationImpl(this, Notification.SET, Java_PackageImpl.MANIFEST_ENTRY__NAME, oldValue, value));
		
	}
	
	
	
	private com.crossecore.ocl.Set<ManifestAttribute> _attributes;
	
	public com.crossecore.ocl.Set<ManifestAttribute> getAttributes()
	{
		if(_attributes==null){
			_attributes = new com.crossecore.ocl.Set<ManifestAttribute>(ManifestAttribute.class, this, Java_PackageImpl.MANIFEST_ENTRY__ATTRIBUTES, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.MANIFEST_ENTRY__ATTRIBUTES);
		}
		return _attributes;
	
	}

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.MANIFESTENTRY;
	}
	
	
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.MANIFEST_ENTRY__NAME:
				return getName();
			case Java_PackageImpl.MANIFEST_ENTRY__ATTRIBUTES:
				return getAttributes();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.MANIFEST_ENTRY__NAME:
				setName((java.lang.String) newValue);
				return;
			case Java_PackageImpl.MANIFEST_ENTRY__ATTRIBUTES:
				getAttributes().clear();
				getAttributes().addAll((java.util.Collection<? extends ManifestAttribute>) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.MANIFEST_ENTRY__NAME:
				return NAME_EDEFAULT == null ? _name != null : !NAME_EDEFAULT.equals(_name);
			case Java_PackageImpl.MANIFEST_ENTRY__ATTRIBUTES:
				return _attributes != null && !_attributes.isEmpty();
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-typescript/java_/ManifestEntryBase.ts
import {EClass} from "ecore/EClass";
import {Set} from "ecore/Set";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {AbstractCollection} from "ecore/AbstractCollection";
import {Bag} from "ecore/Bag";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {ManifestEntry} from "./ManifestEntry";
import {ManifestAttribute} from "./ManifestAttribute";
		
			export class ManifestEntryBase
			extends BasicEObjectImpl
			implements ManifestEntry
			{
				private _name:string = '';
				get name():string{
					return this._name;
				}
				set name(value:string){
					this._name = value; 
				}
				private _attributes:Set<ManifestAttribute> = null;
				
				get attributes():Set<ManifestAttribute>{
					if(this._attributes===null){
						this._attributes = new Set<ManifestAttribute>(this, Java_PackageLiterals.MANIFESTENTRY_ATTRIBUTES, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.MANIFESTENTRY_ATTRIBUTES);
							
					}
					return this._attributes;
					
				}
				
				

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return ManifestEntryBase.eStaticClass;
				}
			
			
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.MANIFESTENTRY_NAME:
							return this.name;
						case Java_PackageLiterals.MANIFESTENTRY_ATTRIBUTES:
							return this.attributes;
					}
					//return this.eGetFromBasicEObjectImpl(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
						case Java_PackageLiterals.MANIFESTENTRY_NAME:
							this.name = <string> newValue;
							return;
						case Java_PackageLiterals.MANIFESTENTRY_ATTRIBUTES:
							this.attributes.clear();
							this.attributes.addAll(newValue);
							return;
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
